devnet-sdk/testing/testlib/validators/forks.go

```diff
@@ -0,0 +1,137 @@
+	return acquireForkConfig(chainIdx, forkName, true)
+}
+
+// RequiresNotFork returns a validator that ensures a specific L2 chain does not have a specific fork activated.
+func AcquireRequiresNotL2Fork(chainIdx uint64, forkName rollup.ForkName) (ChainConfigGetter, systest.PreconditionValidator) {
+	return acquireForkConfig(chainIdx, forkName, false)
+}
+
+// exactForkValidator is a helper function that checks if a specific L2 chain is exactly at a given fork.
+func exactForkValidator(chainIdx uint64, forkName rollup.ForkName, chainConfigMarker interface{}) systest.PreconditionValidator {
+	return func(t systest.T, sys system.System) (context.Context, error) {
+		forkConfig, timestamp, err := getForkConfig(t, sys, chainIdx)
+		if err != nil {
+			return nil, err
+		}
+
+		currentFork := forkConfig.CurrentFork(*timestamp)
+		if currentFork != forkName {
+			return nil, fmt.Errorf("L2 chain %d is at fork %s, but should be exactly at fork %s for this validator to pass", chainIdx, currentFork, forkName)
+		}
+
+		return context.WithValue(t.Context(), chainConfigMarker, forkConfig), nil
+	}
+}
+
+// AcquireRequiresExactL2Fork returns a validator that ensures a specific L2 chain is exactly at a specific fork,
+// meaning the specified fork must be active but no later forks may be activated.
+func AcquireRequiresExactL2Fork(chainIdx uint64, forkName rollup.ForkName) (ChainConfigGetter, systest.PreconditionValidator) {
+	chainConfigMarker := new(byte)
+	validator := exactForkValidator(chainIdx, forkName, chainConfigMarker)
+	return func(ctx context.Context) system.ChainConfig {
+		return ctx.Value(chainConfigMarker).(system.ChainConfig)
+	}, validator
+}
+
+// interopValidator is a helper function that checks if a specific L2 chain has interop enabled or disabled.
+func interopValidator(chainIdx uint64, shouldBeEnabled bool, forkConfigMarker interface{}) systest.PreconditionValidator {
+	return func(t systest.T, sys system.System) (context.Context, error) {
+		forkConfig, timestamp, err := getForkConfig(t, sys, chainIdx)
+		if err != nil {
+			return nil, err
+		}
+
+		isEnabled := forkConfig.IsInteropEnabled(*timestamp)
+		if isEnabled != shouldBeEnabled {
+			if shouldBeEnabled {
+				return nil, fmt.Errorf("L2 chain %d does not have interop enabled, which it should be for this validator to pass", chainIdx)
+			} else {
+				return nil, fmt.Errorf("L2 chain %d has interop enabled, but it should not be for this validator to pass", chainIdx)
+			}
+		}
+
+		return context.WithValue(t.Context(), forkConfigMarker, forkConfig), nil
+	}
+}
+
+// AcquireInteropConfig returns a ForkConfigGetter and a PreconditionValidator
+// that ensures a ForkConfig is available for the specified L2 chain and checks
+// if interop is enabled or disabled as specified.
+func AcquireInteropConfig(chainIdx uint64, shouldBeEnabled bool) (ChainConfigGetter, systest.PreconditionValidator) {
```

Comment:
My opinion is that this can be removed, users can call `AcquireRequiresL2Fork` with the fork name.

Reply:
See above

Reply:
Removing